### DISCUSSION TITLE
Lost types of loader after upgrading to 2.9.0

### DISCUSSION THREAD

**Discussion by merthanmerter** [remix-run/remix#9297] [Q&A] [answered]
lost return types of loader after upgrading to 2.0.0. Am I missing something?

`const { data } = useLoaderData<typeof loader>();`

**Comment by gustavopch** [comment, marked as answer]
Same as https://github.com/remix-run/remix/issues/9295.

**Reply by merthanmerter** [reply]
didn't want to create an issue if i was missing something. but thanks!

~~Weird thing is, downgrading also didn't work.~~
